src/GenericSpectator/Assets/Scripts/PlatformDeviceBootstrapper.cs

```diff
@@ -0,0 +1,49 @@
+﻿using UnityEngine;
+
+public class PlatformDeviceBootstrapper : MonoBehaviour
```

Comment:
>PlatformDeviceBootstrapper [](start = 13, length = 26)

@matthejo, I didn't want to have to build a different scene for each platform, so I made a single scene with a prefab instantiator for the specific platform's device (camera+goo) stuff.  Any concerns with this approach?

When the ARCore and/or ARKit SDK folders are hidden, the corresponding prefab has missing scripts in it, but as long as I don't instantiate the prefab, the console/error log doesn't get spammed with "missing script" warnings, so that's fortunate.

Reply:
FWIW, this is how we have modeled some other platform specific components. ITrackingObservers are instantiated in this manner.

Reply:
Cool.  Good to know!

---
In reply to: [337246014](https://github.com/microsoft/MixedReality-SpectatorView/pull/250#discussion_r337246014) [](ancestors = 337246014)